promscale: 0.10.0 -> 0.17.0

anpin opened the pull request:
###### Description of changes
1. promscale was updated from `0.10.0` to `0.17.0`
2. which required adding separate `promscale_extension` for `postgresql` built using  `buildPgxExtension` function developed [here](https://github.com/NixOS/nixpkgs/pull/186276)
3. Added myself to maintainers 

P.S. This PR was prepared before the `promscale` team [announced](https://github.com/timescale/promscale/issues/1836) deprecation of the project, but since it was already made I'm submitting it anyways.

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [X] Tested, as applicable:
  - [NixOS test(s)](https://github.com/NixOS/nixpkgs/blob/556fcb7fcb35318ce987c4138355035d078cb614/nixos/tests/promscale.nix) for `postgresql` 12, 13, 14 
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

risicle commented:
@ofborg eval

risicle commented on pkgs/servers/monitoring/prometheus/promscale/default.nix:
```diff
@@ -7,28 +7,27 @@
   ];
-
+  preBuild = ''
+    # main module (github.com/timescale/promscale) does not contain package github.com/timescale/promscale/migration-tool/cmd/prom-migrator
```
I don't understand this comment.

0x4A6F commented:
Thanks.

Could you format your commit message according to [CONTRIBUTIONS.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).
There are also 10 commits changing 7 files, that can be condensed into (e.g.):

- maintainers: add anpin
- promscale: 0.10.0 -> 0.17.0
- postgresqlPackages.promscale_extension: init at 0.8.0
- nixosTests.promscale: add tests for promscale_extension

anpin committed:
maintainers: add anpin

anpin committed:
postgresqlPackages.promscale_extension: init at 0.8.0

anpin committed:
promscale: 0.10.0 -> 0.17.0

anpin committed:
nixosTests.promscale: add tests for promscale_extension

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2046

SuperSandro2000 commented on pkgs/servers/sql/postgresql/packages.nix:
```diff
@@ -75,4 +75,7 @@ self: super: {
 
     wal2json = super.callPackage ./ext/wal2json.nix { };
+
+    promscale_extension = super.callPackage ./ext/promscale_extension.nix { };
```
Please sort this

SuperSandro2000 commented on pkgs/servers/sql/postgresql/ext/promscale_extension.nix:
```diff
@@ -0,0 +1,60 @@
+  ];
+
+  preBuild = ''
+    patchShebangs create-upgrade-symlinks.sh
+    patchShebangs extract-extension-version.sh
```
```suggestion
    patchShebangs create-upgrade-symlinks.sh extract-extension-version.sh
```

SuperSandro2000 commented on pkgs/servers/sql/postgresql/ext/promscale_extension.nix:
```diff
@@ -0,0 +1,60 @@
+  preBuild = ''
+    patchShebangs create-upgrade-symlinks.sh
+    patchShebangs extract-extension-version.sh
+    ## A hack to boostrap the build, some pgx commands require this file. It gets re-generated later.
```
```suggestion
    # Hack to boostrap the build because some pgx commands require this file. It gets re-generated later.
```

SuperSandro2000 commented on pkgs/servers/sql/postgresql/ext/promscale_extension.nix:
```diff
@@ -0,0 +1,60 @@
+    # there is a duplicate definition in the lock file which fails to build with buildRustPackage
+    (fetchpatch {
+      name = "cargo-vendor.patch";
+      url = "https://patch-diff.githubusercontent.com/raw/timescale/promscale_extension/pull/611.patch";
```
```suggestion
      url = "https://github.com/timescale/promscale_extension/commit/3cef3f26f72ebf52d8800910ea655cac09312c57.patch";
```
we want to use an url which does not change if more things are pushed to the PR

0x4A6F reviewed (approved):
LGTM, but remember to remove unsupported software from nixpkgs.
A warning should also be thrown, might be implemented with `lib.warn`:

```diff
diff --git a/pkgs/servers/monitoring/prometheus/promscale/default.nix b/pkgs/servers/monitoring/prometheus/promscale/default.nix
index 7dcc5f01e47..cfdb1756128 100644
--- a/pkgs/servers/monitoring/prometheus/promscale/default.nix
+++ b/pkgs/servers/monitoring/prometheus/promscale/default.nix
@@ -52,5 +52,6 @@ buildGoModule rec {
     license = licenses.asl20;
     platforms = platforms.unix;
     maintainers = with maintainers; [ _0x4A6F anpin ];
+    broken = lib.warn "promscale is deprecated and will be removed in a future release, see https://github.com/timescale/promscale/issues/1836" false;
   };
 }
```

SuperSandro2000 commented:
> LGTM, but remember to remove unsupported software from nixpkgs.
> A warning should also be thrown, might be implemented with `lib.warn`:

This needs to be reverted because it does not go well with ofborg

anpin committed:
promscale_extension: fixed cargoPatch
somehow the patch made earlier stoped working, so the build was failing
due to git dependency in cargo.lock. Fixed that in the upstream PR and
repointed fetchpatch to the new commit

marsam reviewed (approved):
LGTM. Thank you!

3noch commented:
> > LGTM, but remember to remove unsupported software from nixpkgs.
> > A warning should also be thrown, might be implemented with `lib.warn`:
> 
> This needs to be reverted because it does not go well with ofborg

How can we add this warning back? It's important to communicate this to users of this package.

SuperSandro2000 commented:
We can't unless it shouldn't be cached on cache.nixos.org. Maybe we can rewrite a release note entry but then we should also remove it probably.

3noch commented:
It's already not cached because it's `unfree` license.
